pkgs/dart_model/lib/src/scopes.dart

```diff
@@ -157,10 +157,12 @@ class MacroScope {
 
   /// Merges a new partial [model] into the scope's accumulated model spanning
   /// multiple queries.
-  // TODO: Actually accumulate instead of replacing the model
   void addModel(Model model) {
-    _accumulatedModel = model;
-    _typeSystem = null;
+    if (_accumulatedModel case var accumulated?) {
+      accumulated.mergeWith(model);
```

Comment:
I think that should be `_accumulatedModel = ...` :)

I guess we don't have any test coverage for this :) meaning we never use the type system after a second query.

Reply:
hah!

Reply:
https://github.com/dart-lang/macros/pull/110